### ISSUE TITLE
Potential configuration options

### ISSUE THREAD

**Issue by ryangjchandler** [ryangjchandler/prettier-plugin-blade#26]
We should start thinking about some of the configuration options that we're going to want to use now before we get too deep into implementation details.

- `tabWidth` - `4`
- `spaceAfterDirectiveName` - `false`

**Comment by olivernybroe** [comment]
Let's make `spaceAfterDirectiveName` into `spacesAfterDirectiveName - 0`, that way for people who prefer 2 spaces, it's a simple change. 

And it regards to code, we can just add the amount of spaces based on the number.

**Comment by olivernybroe** [comment]
Another configuration option I think we should have would be `indentBetweenDirective - true`, which defines if statements between ex. `if`, `endif` should be indented or not.

**Comment by ryangjchandler** [comment]
Yeah, makes sense!

**Comment by dylanmichaelryan** [comment]
Maybe something like `multiLineProps`

example

If set to true
```
@props([
    'title',
    'description',
])
```


If set to false
```
@props(['title', 'description'])
```

**Comment by ryangjchandler** [comment]
> Maybe something like `multiLineProps`
> 
> 
> 
> example
> 
> 
> 
> If set to true
> 
> ```
> 
> @props([
> 
>     'title',
> 
>     'description',
> 
> ])
> 
> ```
> 
> 
> 
> 
> 
> If set to false
> 
> ```
> 
> @props(['title', 'description'])
> 
> ```

This one might be trickier since the directive content is formatted by the prettier-php plugin. If that's configurable the we can do it!

**Comment by olivernybroe** [comment]
One thing to be aware of is that prettier is a really opinionated formatter, with close to no configurations. 

I think we should be careful about not changing their philosophy too much 🤔 
https://prettier.io/docs/en/option-philosophy.html

**Comment by ryangjchandler** [comment]
Yeah, I agree here. Having configuration options for spaces after directive names is probably all we want to begin with.

**Comment by mikebronner** [comment]
> I think we should be careful about not changing their philosophy too much

I don't mean to discount the efficacy of Prettier, or start a flame war, or be controversial in any way. But since prettier is very opinionated in its target to JavaScript, and very few other comprehensive Blade formatters exist, not providing configuration options to allow for PHP/Blade code styles where not covered by PSRs seems that it might undercut the target market of this package. I don't mean to start a discussion, just food for thought. We have not adopted Prettier in our company for this reason so far, and were looking forward to finally having a package that might address our formatting needs.

Which-ever way you choose, I respect your choice as package maintainers, and the goals you set for yourself. Thanks again for your consideration. :)

**Comment by claytonrcarter** [comment]
FWIW I would advocate for having as few configuration options as possible. I would vote against even having a `spaceAfterDirectiveName` in favor of deferring to the Laravel docs or perhaps how the prettier PHP plugin handles statements and and expressions.

**Comment by zepfietje** [comment]
I totally agree with @claytonrcarter.

`tabWidth` of `4` would probably make sense, since this is used in a new Laravel project as well.

`spaceAfterDirectiveName` is a bit harder, since the Laravel docs and new Laravel projects add a single space after the `@if` and loop directives. However, custom conditional directives as well as all other directives seem to not have a space at all. So probably a bit more to discuss for that one.

In general, I do think having as few configuration options as possible would be beneficial.
